### PR TITLE
Add feature detection for IndexedDB 2.0

### LINKED ISSUE #2571: add support for indexeddb 2.0
https://caniuse.com/#search=indexeddb

Now the indexeddb detect script only detects basic indexeddb apis.

But some browser versions implement 1.0 and some 2.0. 

So we can add a new script to detect 2.0 support.

In my opinion, 2.0 add some new apis compared to 1.0. Such as `getAll/getAllKeys`.

So we can use `'getAll' in IDBIndex.prototype` to determine whether the browser supports it.

### PR DESCRIPTION
**Pull request by Xenonym**
Closes #2571.

### CODE REVIEW

File: feature-detects/indexeddb2.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*!
+{
+  "name": "IndexedDB 2.0",
+  "property": "indexeddb2",
+  "tags": ["storage"],
+  "caniuse": "indexeddb2",
+  "authors": ["Tan Zhen Yong (@Xenonym)"],
+  "polyfills": ["indexeddb"],
```
Comment (rejas) [inline feature-detects/indexeddb2.js:8]: is that polyfill also working for indexedDB 2.0?

File: feature-detects/indexeddb2.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*!
+{
+  "name": "IndexedDB 2.0",
+  "property": "indexeddb2",
+  "tags": ["storage"],
+  "caniuse": "indexeddb2",
```
Comment (rejas) [inline feature-detects/indexeddb2.js:6]: please add this also to the test/browser/integration/caniuse.js file

File: feature-detects/indexeddb2.js (hunk 1 above)
Comment (Xenonym) [reply feature-detects/indexeddb2.js:8]: Yep! IndexedDBShim has 2.0 only methods like [`getAll()` and `getAllKeys()`](https://github.com/indexeddbshim/IndexedDBShim/blob/master/src/IDBIndex.js#L429).

File: feature-detects/indexeddb2.js (hunk 2 above)
Comment (Xenonym) [reply feature-detects/indexeddb2.js:6]: It's already in the PR: https://github.com/Xenonym/Modernizr/blob/feature/add-indexeddb-2/test/browser/integration/caniuse.js#L99

**Comment by rejas** [comment]
Looks good. One last thing though: Could you move the test into the storage subdirectory? seems to fit in there. Yes, indexeddb v1 isnt in there yet but we can clean that up later before v4

**Comment by Xenonym** [comment]
> Could you move the test into the storage subdirectory? 

Moved!

**Comment by Markel** [comment]
Nice to see some hacktoberfest contributions :)

**Comment by rejas** [comment]
any comments from you @Markel about this PR? think it can be merged?

**Comment by Markel** [comment]
Sorry for not getting back, kinda of a busy week with exams (writing in a train right now :)

I like the detect :) It has already been merged so congrats @Xenonym on your first commit  to the repository 🥳🥳 and thank you 😊

Btw @alsotang you may want to check this out :)

**Comment by alsotang** [comment]
the detect method is right. thank you.

Markel F. <notifications@github.com>于2020年10月31日 周六04:07写道：

**Pull request merged**